src/common/latest/sonarqube/Scanner.ts

```diff
@@ -31,17 +32,34 @@ export default class Scanner {
     return this.isSonarCloud;
   }
 
-  static async downloadScanner(fileUrl: string) {
+  static async downloadScanner(fileUrl: string, name: string, version: string) {
+    let cachedPath;
+
+    const semverCompliantVersion = semver.coerce(version)?.version;
+    if (!semverCompliantVersion) {
+      throw new Error(`Invalid version format: ${version}`);
+    }
+
     try {
-      log(LogLevel.DEBUG, `Downloading scanner from ${fileUrl}`);
+      cachedPath = toolLib.findLocalTool(name, semverCompliantVersion);
+      if (cachedPath) {
+        log(LogLevel.DEBUG, `Found cached scanner at ${cachedPath}`);
+        return cachedPath;
+      }
+      log(LogLevel.DEBUG, `Cache not found for ${name} ${semverCompliantVersion}`);
```

Comment:
It feels to me we could make an effort to guide users about how they can:
- Cache the downloaded Scanner
- Cache the Scanner specific cache

It feels like a dedicated effort to me which involves the documentation team, is it worth a dedicated ticket? And I'm wondering if we can make a simple sentence here to guide users towards looking at the `Cache` task

just a thought, feel free to change or ignore:
```suggestion
      log(LogLevel.DEBUG, `Cache not found for ${name} ${semverCompliantVersion}. Are you using the Cache task to cache the path '$(Agent.ToolsDirectory)/${name}'?`);
```

Reply:
agreed, the docs should provide instructions on caching. In the meantime, your comment suggestion is better than nothing (indicates we support caching)